tests/test_headers.py

```diff
@@ -202,6 +202,20 @@ def test_content_length(self, client):
         content_length = str(len(SAMPLE_BODY))
         assert result.headers['Content-Length'] == content_length
 
+    def test_declare_content_length(self, client):
+        resource = testing.SimpleTestResource(headers={'Content-Length': 42})
```

Comment:
This actually just sets the header directly rather than going through the `content_length` property. To test that, you will need to create a new resource class that sets `req.content_length` from within its `on_get()` responder, e.g.:

```py
class SomeTestResource(object):
    def on_get(self, req, resp):
        resp.content_length = 42
```